### PR TITLE
[Backport 0.24] Admin api for brokers

### LINKED ISSUE #5405: Admin api to trigger snapshot in a broker
**Is your feature request related to a problem? Please describe.**

Currently the snapshots are taken only at the configured snapshot interval. It would be useful to expose an api through which we can trigger a snapshot without waiting for the snapshot interval. This is useful 
* For debugging
* For testing
* To mitigate the effects of some bugs that might occur due to reprocessing

In addition, we can also add other functions such as stop the stream processor, query the status of a partition etc. 

**Describe the solution you'd like**

We can expose the interface as a rest endpoint similar to health status or ready status. The users should be able to enable or disable these endpoints via configuration. 

**Additional context**
This feature will be useful to prevent some of the known bugs that may occur during an upgrade. Through this api users can force a snapshot on all partitions before triggering an upgrade.

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

* Enable pausing of streamprocessor
* Backport partitions actuator api from #5408 

## Related issues

closes #5405

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by deepthidevaki** [comment]
@saig0 This is not a simple cherry-pick from the original PR. There were some missing functionalities in 0.24 which needed to be added.

File: qa/integration-tests/src/test/java/io/zeebe/broker/it/system/BrokerAdminServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+  private Broker leader;
+
+  @Before
+  public void before() {
+    leader = clusteringRule.getBroker(clusteringRule.getLeaderForPartition(1).getNodeId());
+    leaderAdminService = leader.getBrokerAdminService();
+  }
+
+  @Test
+  public void shouldReportPartitionStatus() {
```
Comment (deepthidevaki) [inline qa/integration-tests/src/test/java/io/zeebe/broker/it/system/BrokerAdminServiceTest.java:51]: I added this test to test the follower status. It is not in original PR. Once this is reviewed, I will port it to develop.

**Review by saig0 — APPROVED** [review]
Looks good. Thanks @deepthidevaki :apple:

**Pull request merged**